openshift/openshift_transport.go

```diff
@@ -139,6 +103,7 @@ func (ref openshiftReference) DockerReference() reference.Named {
 func (ref openshiftReference) PolicyConfigurationIdentity() string {
 	res, err := policyconfiguration.DockerReferenceIdentity(ref.dockerReference)
 	if res == "" || err != nil { // Coverage: Should never happen, NewReference constructs a valid tagged reference.
+		// TODO(runcom,mitr): it's awful to panic inside a library
```

Comment:
Not… cleanly? What would we the caller do with an identity-less image? Would we fall closed to rejecting it, even if the default policy were `insecureAcceptAnything`? We can’t just use the default because perhaps the image _should_ have an identity which _should_ have matched something else.

And really, this error by construction _cannot happen_. Burdening the API with handling errors which can’t happen is suboptimal when we can hide the ugliness in this single file.

We could either use the type system to demonstrate this (`interface referenceNamedWithExactlyOneOfTagAndDigest`), move the theoretically failing (but never in practice) part into `NewReference`, or just confidently do `res, _ := …` here, knowing that the error cannot happen. Or, well, `panic`. Given that this “cannot happen”, I tend to favor short solutions which don’t spread the ugliness around but still detect that the circumstances have changed; though, I suppose, that `interface` one would be most clear and convincing in demonstrating that everything is fine.